### PR TITLE
Prevent 503s from killing the client during discovery

### PR DESCRIPTION
**Pull request by fabianvf**
Sometimes aggregated API servers are down, which results in a 503 during discovery. Rather than raising an exception during discovery, we'll instead ignore the error, and an error won't be raised until the user attempts to access that API. This allows users to ignore downed APIs that they don't use.

### CODE REVIEW

File: dynamic/discovery.py [hunk 1]
```diff
@@ -155,7 +155,10 @@ def get_resources_for_api_version(self, prefix, group, version, preferred):
         subresources = {}
 
         path = '/'.join(filter(None, [prefix, group, version]))
-        resources_response = self.client.request('GET', path).resources or []
+        try:
+            resources_response = self.client.request('GET', path).resources or []
+        except ServiceUnavailableError:
+            resources_response = []
```
Comment (palnabarun) [inline dynamic/discovery.py]: I think when the exception is caught we should probably log a warning. What do you think?

```
logger.warn('some helpful text')
```

Comment (fabianvf) [reply dynamic/discovery.py]: makes sense, though this module is lacking logging in general. I can add it here and then maybe do a followup PR to add it to rest?

Comment (palnabarun) [reply dynamic/discovery.py:161]: Sounds good to me. :+1:

**Pull request merged**